include/SDL_config_windows.h

```diff
@@ -25,6 +25,9 @@
 
 #include "SDL_platform.h"
 
+/* This defines _WIN32_WINNT for SDK version detection */
+#include <sdkddkver.h>
+
```

Comment:
sdkddkver.h is not available in all build environments. Older Visual Studio installations may or may not have it depending on which components you have installed, and even older Visual Studio versions didn't have it at all.

Reply:
I'm wondering if we actually want `winsdkver.h` here and `_WIN32_MAXVER` instead of `sdkddkver.h` and `_WIN32_WINNT`. The former is the max OS version the SDK supports and the latter is what OS version the application is actually targeting.

Reply:
Possibly. We should check whether that header is available in various build environments too. :)

Reply:
Yep, I'll see if I can get my hands on the XP SDK to see if one or both of those headers are present.

Reply:
Ok, it's a bit of mess going back all the way to XP. What I found in the XP SP2 SDK is that `windows.h` itself is actually what's setting `WINVER` if it's undefined:
```
#ifndef WINVER
#define WINVER 0x0501
#else
#if defined(_WIN32_WINNT) && (WINVER < 0x0400) && (_WIN32_WINNT > 0x0400)
#error WINVER setting conflicts with _WIN32_WINNT setting
#endif
#endif
```

We certainly don't want to be pulling `windows.h` in with `SDL_config.h`, since we might break build for applications that need a particular header include order or need to set some `#defines` before including `windows.h`.

I looked at the Windows 7 SP1 SDK (which can be used to target XP) and found that it does have both `sdkddkver.h` and `winsdkver.h`, likewise with mingw-w64 which has contained winsdkver.h [since 2014](https://github.com/mirror/mingw-w64/commit/0c890e289cdbb3c52a6c9a81e98ed7e6973b5587).

Given all that, I pushed what is probably the best we can hope to do with only the C preprocessor at our disposal: using `winsdkver.h` and providing an opt-out via `NO_WINSDKVER_H`.

While we could just keep the `_MSC_VER` check as is, it seems wrong to handicap all non-MSVC builds (and even some fairly recent VS versions like 2015/2017) today in order to support people building with ancient XP/Vista-era SDKs. Folks using these old SDKs can either set `NO_WINSDKVER_H` or (better yet) update their SDK.

Reply:
Sounds good.